front50-core/src/main/java/com/netflix/spinnaker/front50/model/application/ApplicationService.java

```diff
@@ -171,12 +171,13 @@ private void removeApplicationFromProjects(String appName) {
             p -> {
               log.info("Removing application '{}' from project '{}'", appName, p.getId());
               p.getConfig().getApplications().remove(appName);
-              p.getConfig()
-                  .getClusters()
-                  .forEach(
-                      c ->
-                          Optional.ofNullable(c.getApplications())
-                              .ifPresent(apps -> apps.remove(appName)));
+              Optional.ofNullable(p.getConfig().getClusters())
+                  .ifPresent(
+                      clusters ->
+                          clusters.forEach(
+                              c ->
+                                  Optional.ofNullable(c.getApplications())
+                                      .ifPresent(apps -> apps.remove((appName)))));
```

Comment:
No idea if this is the idiomatic way of doing this in Java. Please let me know if there's a better way!